TEMPLATES.md

```diff
@@ -0,0 +1,103 @@
+# Using `Template`s as Execution Targets With CI Operator
+
+The CI Operator allows for users to configure tests running in a single container
+with the `tests` array. In many cases, however, running a command in a single
+container will not suffice. For more complicated tests, the CI Operator supports
+a `--template` option with which a test author may write black-boxed execution
+targets using OpenShift `Template`s. This document explains the interfaces expected
+for test `Template`s and best practices for how an author should configure their
+`Template`.
+
+The high-level contract for a `Template` execution target is that the `Template`
+will be processed into the test's `Namespace` once prerequisite `Build`s have
+occurred. Artifacts will be gathered from `artifacts` volumes on `Pod`s in
+the `Template`, if any are present. What work occurs in a `Template`'s `Pod`s is
+a black box as far as the CI Operator is concerned.
+
+## Parameters Available to `Template`s
+
+A number of parameters are available to the `Template` and will be provided if
+the `Template` is explicitly configured to consume them. The following parameters
+will be provided to `Template`s:
+
+ - `NAMESPACE`
+   The namespace in which your `Template` is processed. Target this namespace with
+   your objects.
+
+   | Example: | `ci-op-<input-hash>` |
+   | - | - |
+
+ - `IMAGE_FORMAT`
+   The image repository URL template using which your job's release images can
+   be pulled, formatted for the OpenShift Ansible `-e oreg_url` parameter.
+   `Template`s depending on this parameter will be processed only after all
+   images are built.
+
+   | Example: | `registry.svc.ci.openshift.org/ci-op-<input-hash>/stable:${component}` |
+   | - | - |
+
+
+ - `RPM_REPO`
+   The RPM repository URL from which your job's RPMs can be installed.
+   `Template`s depending on this parameter will be processed only after the
+   `rpm` image is built.
+
+   | Example: | `http://rpm-repo-ci-op-<input-hash>.svc.ci.openshift.org` |
+   | - | - |
+
+ - `IMAGE_<component>`
+   The image repository URL from which the `<component>` image can be pulled.
```

Comment:
There's mixing and matching of jargon here; "component" refers to "component image", right? I'd just use "image", and clarify where that comes from. E.g. "`IMAGE_<image>` expands to the image repository URL from which image `<image>` can be pulled. The list of available values for `<image>` is defined in the CI Operator config file under `images:`. Dashes in the config file image name are converted to underscores."